client-runtime/src/main/java/com/microsoft/rest/v2/protocol/HttpResponseDecoder.java

```diff
@@ -283,6 +285,7 @@ private Type getHeadersType() {
         }
 
         if (token.isSubtypeOf(RestResponse.class)) {
+            token = token.getSupertype(RestResponse.class);
```

Comment:
It doesn't look like RestResponse has a supertype/base class. What does this return?

Reply:
In this case `token` is something like `BodyResponse` or `BlobGetResponse` and we're walking up the hierarchy to get the `RestResponse` superclass so that we can look at the type parameters in a uniform way.